src/states/stay/CountdownToStay.java

```diff
@@ -1,7 +1,60 @@
 package states.stay;
 
-import states.Countdown;
+import events.TimerRanOut;
+import events.TimerTicked;
+import events.TimerUp;
+import states.SecurityContext;
+import states.SecurityState;
+import timer.Notifiable;
+import timer.Timer;
 
-public class CountdownToStay extends Countdown {
+public class CountdownToStay extends SecurityState implements Notifiable {
+
+	private static CountdownToStay instance;
+	private Timer timer;
+
+	/**
+	 * Private constructor for the singleton pattern
+	 */
+	protected CountdownToStay() {
+	}
+
+	public static CountdownToStay instance() {
+		if (instance == null) {
+			instance = new CountdownToStay();
+		}
+		return instance;
+	}
+
+	@Override
+	public void enter() {
+	}
+
+	@Override
+	public void leave() {
+	}
+
+	public void startTimer(int seconds) {
+		if (timer == null) {
+			timer = new Timer(this, seconds);
+		}
+	}
+
+	public void stopTimer() {
+		timer = null;
+	}
+
+	public int getTimeValue() {
+		return timer.getTimeValue();
+	}
+
+	public void handleEvent(TimerTicked event) {
+		SecurityContext.instance().showSecondsToStay(timer.getTimeValue());
+	}
+
+	public void handleEvent(TimerRanOut event) {
+		SecurityContext.instance().showSecondsToStay(timer.getTimeValue());
+		SecurityContext.instance().handleEvent(TimerUp.instance());
+	}
```

Comment:
Shouldn't there be a change state here?

Reply:
No, per your previous question, CountdownToStay is my super to CountdownToStayReady and CountdownToStayNotReady. It registers TimeRanOut event, but for its children it throws TimerUp event. It operates the timer on this level so it has to listen to TimerRanOut, but throw a separate event.

Reply:
Okay looks good